## Hanna's Midterm
Data on more than 5,800 clergy who have been listed as credibly accused of sexual abuse in reports released by Catholic dioceses and religious orders from 2002-2020. 

### Import Python tools

In [34]:
import pandas as pd

### Read data

In [35]:
url = 'file:///Users/hannakang/Downloads/clergymembers.html'

In [36]:
df = pd.read_html(url)[0]

### Basic information about data

In [37]:
df

0                 1           2          3   \
0         id              name  first_name  last_name   
1     652690      Gerald Prinz      Gerald      Prinz   
2     652691     Gerard Kinane      Gerard     Kinane   
3     652692   Lawrence Cavell    Lawrence     Cavell   
4     652693    Patrick Kujawa     Patrick     Kujawa   
...      ...               ...         ...        ...   
6749  652685  Claude Boudreaux      Claude  Boudreaux   
6750  652686        Dac Nguyen         Dac     Nguyen   
6751  652687       Dale Guidry        Dale     Guidry   
6752  652688      Daniel Poche      Daniel      Poche   
6753  652689   Etienne LeBlanc     Etienne    LeBlanc   

                                                     4           5         6   \
0                                                status  birth_date  ord_date   
1     All priestly faculties permanently removed by ...         NaN      1968   
2     All priestly faculties permanently removed wit...         NaN      1973   
3     All priestly faculties permanently removed by ...         NaN      1989   
4                  Convicted and currently incarcerated         NaN      1996   
...                                                 ...         ...       ...   
6749                                           Deceased         NaN      1955   
6750                 Current status or location unknown         NaN      1988   
6751                                           Deceased         NaN      1982   
6752  All priestly faculties permanently removed by ...         NaN      1978   
6753  All priestly faculties permanently removed by ...         NaN      1971   

                                                     7   \
0                                           assignments   
1             Served at Saint Francis de Sales in Houma   
2                     Served at Saint Hilary in Mathews   
3                   Served at Holy Cross in Morgan City   
4     Served at Holy Cross in Morgan City and Saint ...   
...                                                 ...   
6749                                                NaN   
6750        Served at Thanh Gia (Holy Family) in Amelia   
6751  Served at Holy Cross in Morgan City and Saint ...   
6752                Served at Holy Cross in Morgan City   
6753                      Served at Annunziata in Houma   

                             8                                9   \
0                    diocese_id                     diocese_name   
1        Saint Gregory in Houma    and Saint Louis in Bayou Blue   
2                          2581       Diocese of Houma-Thibodaux   
3      Sacred Heart in Montegut  Saint Francis de Sales in Houma   
4                          2581       Diocese of Houma-Thibodaux   
...                         ...                              ...   
6749                       2581       Diocese of Houma-Thibodaux   
6750                       2581       Diocese of Houma-Thibodaux   
6751                       2581       Diocese of Houma-Thibodaux   
6752    Sacred Heart in Cut Off        Saint Bernadette in Houma   
6753  Holy Cross in Morgan City             Holy Family in Dulac   

                                                     10  \
0                                             list_name   
1                                                  2581   
2     Disclosure of Names - The following priests ar...   
3                          Saint Genevieve in Thibodaux   
4     Disclosure of Names - The following priests ad...   
...                                                 ...   
6749  Disclosure of Names - The following priests ar...   
6750  Disclosure of Names - The following priests ar...   
6751  Disclosure of Names - The following priests ad...   
6752                    Saint Francis de Sales in Houma   
6753                              Saint Eloi in Theriot   

                                                     11  \
0                                                  slug   

### How many unique counts of sexual assault were reported?

In [49]:
len(df)

6754

### What is the distribution?

In [47]:
df.describe()

0              1     2       3         4     5     6   \
count     6754           6754  6753    6719      6110  1894  3507   
unique    6754           5898   797    4384      1162   298   147   
top     651122  Robert Poandl  John  Murphy  Deceased  1946  1955   
freq         1             10   576      27      2576    49    97   

                      7     8                           9   \
count               3986  6687                        6742   
unique              2270  2766                        2827   
top     Assistant Pastor  2496  Archdiocese of Los Angeles   
freq                  99   301                         301   

                                                       10  \
count                                                6754   
unique                                               2834   
top     CONSOLIDATED LIST OF PERSONS NAMED IN THE REPO...   
freq                                                  297   

                                             11    12  
count                                      6752  6751  
unique                                     5039  5239  
top     WITH CURRENT STATUS THROUGH NOVEMBER 30  2018  
freq                                        297   297

##### = there were 6,754 unique counts of sexual assault, with Robert Poandl recording the highest count of sexual assault combined: 10. 2,576 of the accused clergymembers are deceased, the highest percentage of the current status of these accused clergymembers. Most of them held assistant pastor positions, and 301 of the assaults occurred at the Archdiocese of Los Angeles. 

### when did the clergy get ordinated? (oldest)

In [54]:
df.sort_values('ord_date', ascending=True).head()

KeyError: 'ord_date'